javalin/src/main/java/io/javalin/http/staticfiles/JettyResourceHandler.kt

```diff
@@ -52,6 +59,49 @@ class JettyResourceHandler(val precompressStaticFiles: Boolean = false) : io.jav
         }
     }
 
+    inner class AliasResourceHandler(private val urlPathPrefix: String, private val aliasCheck: AliasCheck) : ResourceHandler() {
+        override fun getResource(path: String): Resource {
+            val targetResource by lazy { path.removePrefix(urlPathPrefix) }
+            return when {
+                urlPathPrefix == "/" -> getResourceWithAliasChecks(path)!!
+                targetResource == "" -> getResourceWithAliasChecks("/")!! // directory without trailing '/'
+                !path.startsWith(urlPathPrefix) -> EmptyResource.INSTANCE
+                !targetResource.startsWith("/") -> EmptyResource.INSTANCE
+                else -> getResourceWithAliasChecks(targetResource)!!
+            }
+        }
+
+        // same as regular ResourceHandler but with alias checks
+        private fun getResourceWithAliasChecks(path: String) : Resource? {
+            try {
+                var resource: Resource? = null
+                if (baseResource != null) {
+                    val caconicalPath = URIUtil.canonicalPath(path)
+                    resource = baseResource.addPath(caconicalPath)
+                    if (resource != null && resource.isAlias && !checkAlias(path, resource)) {
+                        return null
+                    }
+                }
+                if ((resource == null || !resource.exists()) && path.endsWith("/jetty-dir.css")) resource = stylesheet
+                return resource!!
+            } catch (ignored: java.lang.Exception) {
+
+            }
+            return null;
+        }
+
+        private fun checkAlias(path: String?, resource: Resource): Boolean {
+            if (resource.isAlias) {
+                if (aliasCheck.check(path, resource)) {
+                    return true
+                }
+                return false
+            }
+            return true
+        }
```

Comment:
```suggestion
        private fun checkAlias(path: String?, resource: Resource) = when {
            !resource.isAlias -> false
            else -> aliasCheck.check(path, resource)
        }
```

Reply:
Not sure about my suggestion, the current implementation returns `true` every time `resource.isAlias` is false, which seems weird to me. Either way, I think a `when` is cleaner than nested `if`s.

Reply:
The current implementation is modified from original jetty's.
>returns true every time resource.isAlias is false

As I understand it: if resource is not an alias, then no check should be done - return true and pass this resource